assets/helpers/tracking/acquisitions.js

```diff
@@ -0,0 +1,138 @@
+// @flow
+
+// ----- Imports ----- //
+
+import { getQueryParameter } from 'helpers/url';
+import { deserialiseJsonObject } from 'helpers/utilities';
+
+
+// ----- Types ----- //
+
+type AcquisitionABTest = {
+  name: string,
+  variant: string,
+};
+
+export type Acquisition = {|
+  referrerPageviewId: ?string,
+  campaignCode: ?string,
+  referrerUrl: ?string,
+  componentId: ?string,
+  componentType: ?string,
+  source: ?string,
+  abTests: AcquisitionABTest[],
+|};
+
+
+// ----- Setup ----- //
+
+const ACQUISITIONS_PARAM = 'acquisitionData';
+const ACQUISITIONS_STORAGE_KEY = 'acquisitionData';
+
+
+// ----- Campaigns ----- //
+
+const baselineTestPrefix = 'gdnwb_copts_memco_sandc_support_baseline_support';
+
+const campaigns : {
+  [string]: string[],
+} = {
+  baseline_test: [
+    `${baselineTestPrefix}_banner`,
+    `${baselineTestPrefix}_epic`,
+    `${baselineTestPrefix}_liveblog`,
+    `${baselineTestPrefix}_header_become_supporter`,
+    `${baselineTestPrefix}_header_subscribe`,
+    `${baselineTestPrefix}_side_menu_become_supporter`,
+    `${baselineTestPrefix}_side_menu_subscribe`,
+  ],
+};
+
+export type Campaign = $Keys<typeof campaigns>;
+
+
+// ----- Functions ----- //
+
+// Retrieves the user's campaign, if known, from the campaign code.
+function getCampaign(acquisition: Acquisition): ?Campaign {
+
+  const campaignCode = acquisition.campaignCode;
+
+  if (!campaignCode) {
+    return null;
+  }
+
+  return Object.keys(campaigns).find(campaign =>
+    campaigns[campaign].includes(campaignCode),
+  ) || null;
+
+}
+
+// Stores the acquisition data in sessionStorage.
+function storeAcquisition(acquisition: Acquisition): boolean {
+
+  try {
+
+    const serialised = JSON.stringify(acquisition);
+    window.sessionStorage.setItem(ACQUISITIONS_STORAGE_KEY, serialised);
+
+    return true;
+
+  } catch (err) {
+    return false;
+  }
+
+}
+
+// Reads the acquisition data from sessionStorage.
+function readAcquisition(): ?Acquisition {
+
+  const stored = window.sessionStorage.getItem(ACQUISITIONS_STORAGE_KEY);
+  return stored ? deserialiseJsonObject(stored) : null;
+
+}
+
+// Builds the acquisition object from data and other sources.
+function buildAcquisition(acquisitionData: Object = {}): Acquisition {
+
+  const referrerPageviewId = acquisitionData.referrerPageviewId ||
+    getQueryParameter('REFPVID') ||
+    null;
+
+  const campaignCode = acquisitionData.campaignCode ||
+    getQueryParameter('INTCMP') ||
+    null;
+
+  return {
+    referrerPageviewId,
```

Comment:
Should this (and campaign code) be named, e.g. `referrerPageViewId: referrer...`?

Reply:
This is an es6 [shorthand](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Property_definitions).